### PR TITLE
Refactor some enrollment manager tests

### PR DESCRIPTION
**Pull request by AndrejMitrovic**

### CODE REVIEW

File: source/agora/consensus/EnrollmentManager.d [hunk 1]
```diff
@@ -965,36 +965,39 @@ unittest
     auto utxo_hash1 = utxo_hashes[0];
     auto enrollment = man.createEnrollment(utxo_hash1);
     assert(man.pool.add(enrollment, &storage.findUTXO));
-    assert(man.getValidatorCount(block_height) + 1 == 1);
+    assert(man.getValidatorCount(block_height) == 0);  // not active yet
```
Comment (Geod24) [inline source/agora/consensus/EnrollmentManager.d:966]: Hum, if it's using  the test genesis  block, is that correct ?

Comment (AndrejMitrovic) [reply source/agora/consensus/EnrollmentManager.d:966]: There's no ledger here that loads the enrollment manager with the validator set.

**Review by Geod24 — APPROVED** [review]
Now I remember why the `+ 1` was in place: It was the number of entries in the pool. But I understand why it would be confusing, and this adds a few more asserts, so merge at will.

**Comment by AndrejMitrovic** [comment]
Yeah but unfortunately that doesn't actually test the amount of items in the pool. It just assumes the count. :/

**Review by linked0 — APPROVED** [review]

**Pull request merged**